internal/querycoordv2/observers/collection_observer.go

```diff
@@ -192,14 +190,14 @@ func (ob *CollectionObserver) observeCollectionLoadStatus(collection *meta.Colle
 			)
 		}
 
-		updated.LoadPercentage = int32(loadedCount * 100 / targetNum)
+		updated.LoadPercentage = int32(loadedCount * 100 / targetNum * int(collection.GetReplicaNumber()))
```

Comment:
Use int32 here may lose precision when target number is greater than 100. So a loaded segment may not change progress at all.

Reply:
After #20999 this shall be resolved